docs: add link to project board to PRs (#9630)
NixOS/nix: .github/PULL_REQUEST_TEMPLATE.md

<!-- Large change: Provide instructions to reviewers how to read the diff. -->

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

> The Nix Team uses [this project board](https://github.com/orgs/NixOS/projects/19) to track and schedule reviews.

Reviewer: ```suggestion
The Nix maintainer team uses [this project board](https://github.com/orgs/NixOS/projects/19) to track and schedule reviews.
```

> The Nix maintainer team uses [this project board](https://github.com/orgs/NixOS/projects/19) to track and schedule reviews.

Reviewer: We may want to also add a note on the process we use for that. Feel free to disagree and merge as is though:

```suggestion
The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [track and schedule reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol). 
```

> The Nix Team uses [this project board](https://github.com/orgs/NixOS/projects/19) to help track and schedule work. This should provide visibility into the initial discussion, assignments, as well as postponments.

Reviewer: ```suggestion
The Nix Team uses [this project board](https://github.com/orgs/NixOS/projects/19) to track and schedule reviews.
```
